src/util/chunk_fs.js

```diff
@@ -78,7 +80,10 @@ class ChunkFS extends stream.Transform {
             }
         } catch (error) {
             console.error('ChunkFS _flush_buffers failed', this.q_size, this._total_num_buffers, error);
-            return callback(error);
+            if (callback) {
+                return callback(error);
+            }
+            this.emit('error', error);
```

Comment:
This is called from _transform and is awaited for.
So I think I would expect that when this function does not receive a callback, it should be expected to return a promise rejection. Since this is async function - we can simply throw. 
```suggestion
            throw error;
```

Reply:
ok, fixed!